ui/StatusQ/src/networkchecker.cpp

```diff
@@ -0,0 +1,62 @@
+#include "StatusQ/networkchecker.h"
+
+NetworkChecker::NetworkChecker(QObject* parent)
+    : QObject(parent)
+    , online(false)
+{
+    connect(&manager, &QNetworkAccessManager::finished, this, &NetworkChecker::onFinished);
+    connect(&timer, &QTimer::timeout, this, &NetworkChecker::checkNetwork);
+
+    updateRegularCheck(active);
+}
+
+bool NetworkChecker::isOnline() const
+{
+    return online;
+}
+
+void NetworkChecker::checkNetwork()
+{
+    QNetworkRequest request(QUrl("http://example.com"));
```

Comment:
Quoting from the `example.com`...

`We provide a web service on the example domain hosts to provide basic information on the purpose of the domain. These web services are provided as best effort, but are not designed to support production applications. While incidental traffic for incorrectly configured applications is expected, please do not design applications that require the example domains to have operating HTTP service.`

Reply:
There is already this isOnline flag that's based on the number of peers. https://github.com/status-im/status-desktop/blob/master/src/app/modules/main/view.nim#L270

Wouldn't it do the trick?

Reply:
or another idea.. The WC feature depends on the WC services, right? In this case WDYT of pinging their service instead?

Reply:
E.g. Windows OS API has some funcs to check if OS is online.  [Raymond Chen](https://devblogs.microsoft.com/oldnewthing/20221115-00/?p=107399) . Could we use their webservices http://www.msftconnecttest.com/connecttest.txt . If not, then in the comments to that post there are links to identical Fedora services